cmd/snap-failure/cmd_snapd.go

```diff
@@ -129,20 +129,38 @@ func (c *cmdSnapd) Execute(args []string) error {
 	prevRev, err := prevRevision("snapd")
 	switch err {
 	case errNoSnapd:
+		logger.Noticef("no snapd snap in the system")
 		// the snapd snap is not installed
 		return nil
 	case errNoPrevious:
-		// this is the first revision of snapd to be installed on the
-		// system, either a remodel or a plain snapd installation, call
-		// the snapd from the core snap
-		snapdPath = filepath.Join(dirs.SnapMountDir, "core", "current", "/usr/lib/snapd/snapd")
-		if !osutil.FileExists(snapdPath) {
-			// it is possible that the core snap is not installed at
-			// all, in which case we should try the snapd snap
-			snapdPath = filepath.Join(dirs.SnapMountDir, "snapd", "current", "/usr/lib/snapd/snapd")
+		logger.Noticef("no previous revision of snapd installed in the system")
+		// snapd is being installed for the first time, or there is simply a single
+		// revision of snapd in the system
+		snapdFromCurrentCorePath := filepath.Join(dirs.SnapMountDir, "core", "current", "/usr/lib/snapd/snapd")
+		snapdFromCurrentSnapdPath := filepath.Join(dirs.SnapMountDir, "snapd", "current", "/usr/lib/snapd/snapd")
+
+		if !release.OnClassic {
```

Comment:
Does this need to consider release.CoreDesktop too?

Reply:
no, from this POV CoreDesktop is a Core system, snapd comes from the snapd snap, and never came from the core snap

Reply:
right i see, so the `else` clause inside here is just as valid here, thanks